### PR TITLE
[IMP] membership: work on duration, not on date

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Task: https://www.odoo.com/web#id=6806&view_type=form&model=project.task&action=333&active_id=131
Pad: https://pad.odoo.com/p/r.yIMRe1VVusFuPpnV

### CODE REVIEW

File: addons/membership/membership.py [hunk 1]
```diff
@@ -458,11 +482,9 @@ def fields_view_get(self, cr, user, view_id=None, view_type='form', context=None
     _inherit = 'product.template'
     _columns = {
         'membership': fields.boolean('Membership', help='Check if the product is eligible for membership.'),
-        'membership_date_from': fields.date('Membership Start Date', help='Date from which membership becomes active.'),
-        'membership_date_to': fields.date('Membership End Date', help='Date until which membership remains active.'),
+        'membership_duration': fields.integer('Membership Duration', help='Duration until which membership remains active.'),
```
Comment (bba-odoo) [inline addons/membership/membership.py]: done

File: addons/membership_sale/models/membership_sale.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    'date_to': end_date,
+                    'state': 'waiting',
+                    'sale_order_line_id': line.id
+                })
+        return True
+
+    @api.one
+    def button_confirm(self):
+        res = super(sale_order_line, self).button_confirm()
+        self._create_membership_line()
```
Comment (ara-odoo) [inline addons/membership_sale/models/membership_sale.py]: need to re-factor?? write function prepare membership line and than create membership line

File: addons/membership_sale/models/membership_sale.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    sale_order_line_id = fields.Many2one('sale.order.line', 'Sale Order line', readonly=True)
+    sale_order_id = fields.Many2one(string='Sale Order', related='sale_order_line_id.order_id')
+
+class account_invoice_line(models.Model):
+    _inherit = 'account.invoice.line'
+
+    @api.one
+    def _prepare_domain_invoice_membership(self, line):
+        domain = super(
```
Comment (ara-odoo) [inline addons/membership_sale/models/membership_sale.py]: typo `_prepare_domain_invoice_membreship` > `_prepare_domain_invoice_membership`

File: addons/membership/membership.py [hunk 4]
```diff
@@ -486,36 +511,43 @@ def action_cancel(self, cr, uid, ids, context=None):
...
                                     'partner': line.invoice_id.partner_id.id,
                                     'membership_id': line.product_id.id,
                                     'member_price': line.price_unit,
-                                    'date': time.strftime('%Y-%m-%d'),
+                                    'date': fields.date.today(),
                                     'date_from': date_from,
                                     'date_to': date_to,
                                     'account_invoice_line': line.id,
                                     }, context=context)
+                    res[line.id] = member_line_obj
```
Comment (ara-odoo) [inline addons/membership/membership.py]: why? I think no need of this line.

File: addons/membership_sale/models/membership_sale.py (hunk 2 above)
Comment (bba-odoo) [inline addons/membership_sale/models/membership_sale.py]: done

File: addons/membership_sale/models/membership_sale.py (hunk 3 above)
Comment (bba-odoo) [inline addons/membership_sale/models/membership_sale.py]: done

File: addons/membership/membership.py (hunk 4 above)
Comment (bba-odoo) [inline addons/membership/membership.py]: done

File: addons/membership/__openerp__.py [hunk 5]
```diff
@@ -45,14 +45,18 @@
...
         'membership_demo.xml',
         'membership_demo.yml'
     ],
     'website': 'https://www.odoo.com/page/community-builder',
+    'qweb': [
+        "static/src/xml/state_membership.xml",
+    ],
     'test': ['test/test_membership.yml'],
     'installable': True,
     'auto_install': False,
```
Comment (ara-odoo) [inline addons/membership/__openerp__.py]: `auto_install=False` is the default

File: addons/membership_sale/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+{
+    'name': 'Membership Sale',
+    'version': '0.1',
```
Comment (ara-odoo) [inline addons/membership_sale/__openerp__.py]: version 0.1?

File: addons/membership/membership_view.xml [hunk 7]
```diff
@@ -217,6 +221,70 @@
 
         <!-- PARTNERS -->
 
+        <record model="ir.ui.view" id="membership_expired_form">
+            <field name="name">Expired Memberships</field>
+            <field name="model">membership.membership_line</field>
+            <field name="arch" type="xml">
+                <form string="Expired Memberships" version="7.0" create="false" edit="false" delete='false'>
+                    <group>
```
Comment (ara-odoo) [inline addons/membership/membership_view.xml]: version="7.0"?

File: addons/membership_sale/__openerp__.py (hunk 6 above)
Comment (bba-odoo) [inline addons/membership_sale/__openerp__.py]: done! replace with 1.0

File: addons/membership/membership_view.xml (hunk 7 above)
Comment (bba-odoo) [inline addons/membership/membership_view.xml]: done

**Pull request closed without merging**